spec/models/manageiq/providers/azure/cloud_manager/refresh_parser_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+describe ManageIQ::Providers::Azure::CloudManager::RefreshParser do
+  let(:ems) do
+    ExtManagementSystem.create_with(
+      :subscription => '--SUBSCRIPTION_ID--',
+      :uid_ems      => '--TENANT_ID--',
+      :enabled      => true,
+    ).find_or_create_by!(
+      :name            => 'Azure',
+      :type            => 'ManageIQ::Providers::Azure::CloudManager',
+      :provider_region => 'westeurope',
+    ) do |ems|
+      MiqServer.seed
+      ems.zone = Zone.first
+      ems.authentications << FactoryGirl.create(:authentication,
+                                                :userid   => '--CLIENT_ID--',
+                                                :password => '--CLIENT_SECRET--',
+                                                :authtype => 'default',
+                                                :type     => 'AuthUseridPassword',)
+    end
+  end
+
+  context 'labels and tags' do
+    let(:vm) do
+      VCR.use_cassette(described_class.name.underscore,
+                       :allow_unused_http_interactions => true) { EmsRefresh.refresh(ems) }
+      ems.vms.find_by(:name => 'az-test')
+    end
+
+    it 'fetches labels' do
+      expect(vm.labels.map { |l| [l.name, l.value] }.to_h).to eq('redhat' => 'brno', 'test' => 'hohoho')
```

Comment:
@AlexanderZagaynov - thinking of readability, can this be simplified to:

`let (:cust_attr) do
    vm.labels.first
end`

expect(cust_attr.name).to eq('redhat')
expect(cust_attr.value).to eq('brno')